workers/unity/Assets/Fps/Scripts/WorldTiles/MapTemplate.cs

```diff
@@ -0,0 +1,113 @@
+
+        public GameObject GetTileForLocation(Vector2 location, Random random)
+        {
+            TileTypeCollection collection = defaultTileCollection;
+
+            // Offset location so 0,0 is the middle of the bitmap
+            location += new Vector2(templateBitmap.width / 2.0f - 1, templateBitmap.height / 2.0f - 1);
+            location /= unitSize;
+            Vector2Int texelLocation = Vector2Int.FloorToInt(location);
+
+            if (texelLocation.x >= 0 && texelLocation.x < templateBitmap.width &&
+                texelLocation.y >= 0 && texelLocation.y < templateBitmap.height)
+            {
+                Color32 index = ColorToLookup(templateBitmap.GetPixel(texelLocation.x, texelLocation.y));
+                if (index.a == 0xff)
+                {
+                    if (!tileLookup.TryGetValue(index, out collection))
+                    {
+                        Debug.Log($"Unknown color: {index}\nThere are {tileLookup.Count} colors available.");
+                        PrintColors();
+                        collection = defaultTileCollection;
+                    }
+                }
+            }
+
+            return collection.GetRandomTile(random);
+        }
+
+        private void OnEnable()
+        {
+            InitializeLookup();
+        }
+
+        private void PrintColors()
+        {
+            var message = tileLookup.Select(element => $"[{element.Key}]")
+                .Aggregate("", (current, next) => $"{current} {next}");
+            Debug.Log(message);
+        }
+
+        private void OnValidate()
+        {
+            // Check for issues
+            if (!templateBitmap.isReadable)
+            {
+                Debug.LogError($"template bitmap {templateBitmap.name} must be Read/Write enabled.");
+            }
+
+            if (templateBitmap.format != TextureFormat.RGBA32)
+            {
+                Debug.LogError($"template bitmap {templateBitmap.name} must be RGBA32.");
+            }
+
+            if (defaultTileCollection == null)
+            {
+                Debug.LogError($"Default Tile Collection must be set.");
+            }
+
+            if (unitSize == 0)
+            {
+                unitSize = 1;
+            }
+
+            InitializeLookup();
+        }
+
+        private void InitializeLookup()
+        {
+            // Fill lookup map
+            tileLookup.Clear();
+            if (tileCollections != null && tileCollections.Length > 0)
+            {
+                foreach (var tileType in tileCollections)
+                {
+                    if (tileType == null)
+                    {
+                        continue;
+                    }
+
+                    var index = ColorToLookup(tileType.DisplayColor);
+                    if (tileLookup.ContainsKey(index))
+                    {
+                        Debug.LogError($"Tile collection with color {ColorUtility.ToHtmlStringRGBA(tileType.DisplayColor)} already exists.");
```

Comment:
should they just skip it?

Reply:
This way it does skip it, but also informs that something went wrong